lib/scripting/post-process/expr.ts

```diff
@@ -17,138 +17,271 @@
-	const operator = result[2].text as BinaryOperator;
-	const rightExpr = result[4];
-	return estree.binaryExpression('+', leftExpr, rightExpr);
+function ppParenthesizedExpression(node: ESIToken): any {
+	return node.children[1].estree;
 }
 
-export function is(result: [Expression, null, Token, null, Expression]): Expression {
-	const leftExpr = result[0];
-	const rightExpr = result[4];
-	return estree.binaryExpression('==', leftExpr, rightExpr);
+function ppMemberAccessExpression(node: ESIToken) {
+	return identifier('.' + node.children[1].estree.name);
 }
 
-export function isNot(result: [Expression, null, Token, null, Token, null, Expression]): Expression {
-	const leftExpr = result[0];
-	const rightExpr = result[6];
-	return estree.binaryExpression('!=', leftExpr, rightExpr);
+function ppSubExpression(node: ESIToken): any {
+	let id: any = null;
+	const argLists = [];
+	let expr = null;
+	let args;
+	for (const child of node.children) {
+		if (child.type === 'MemberAccessExpression' || child.type === 'SimpleNameExpression') {
+			id = child.estree;
+		} else if (child.type === 'OpenParenthesis') {
+			args = [];
+		} else if (child.type === 'ArgumentList') {
+			args = child.estree;
+		} else if (child.type === 'CloseParenthesis') {
+			argLists.push(args);
+		} else if (child.type === 'SubExpression') {
+			expr = child.estree;
+		}
+	}
+	let estree: any;
+	if (argLists.length > 0) {
+		for (const argList of argLists) {
+			estree = callExpression(estree ? estree : id, argList);
+		}
+	} else {
+		estree = id;
+	}
+	if (expr != null) {
+		let found = false;
+		estree = replace(expr, {
+			enter: astNode => {
+				if (!found) {
+					if (astNode.type === 'Identifier') {
+						found = true;
+						return memberExpression(estree, identifier(astNode.name.substr(1)));
+					}
+				}
+			},
+		});
+	}
+	return estree;
 }
 
-export function gte(result: [Expression, null, Token, null, Expression]): Expression {
-	const leftExpr = result[0];
-	const rightExpr = result[4];
-	return estree.binaryExpression('>=', leftExpr, rightExpr);
+function ppInvocationExpression(node: ESIToken): any {
+	let id: any = null;
+	const argLists = [];
+	let expr = null;
+	for (const child of node.children) {
+		if (child.type === 'MemberAccessExpression' || child.type === 'SimpleNameExpression') {
+			id = child.estree;
+		} else if (child.type === 'ArgumentList') {
+			argLists.push(child.estree);
+		} else if (child.type === 'InvocationExpression') {
+			expr = child.estree;
+		}
+	}
+	let estree: any;
+	if (argLists.length > 0) {
+		for (const argList of argLists) {
+			estree = callExpression(estree ? estree : id, argList);
+		}
+	} else {
+		estree = id;
+	}
+	if (expr != null) {
+		let found = false;
+		estree = replace(expr, {
+			enter: astNode => {
+				if (!found) {
+					if (astNode.type === 'Identifier') {
+						found = true;
+						return memberExpression(estree, identifier(astNode.name.substr(1)));
+					}
+				}
+			},
+		});
```

Comment:
Seems like the same block like above. Potential to extract to a method?

Reply:
yes, invocation statements are like subexpressions, but without the initial parens.

`BallRelease.Kick 1,2,x(1,2)` vs `x=BallRelease.Kick(1,2,x(1,2))`